lib/LedgerSMB/Database/Change.pm

```diff
@@ -149,6 +154,7 @@ Returns true if the current sha matches one that has been applied.
 
 sub is_applied {
     my ($self, $dbh) = @_;
+    $dbh->clone({ AutoCommit => 1}); # Clone to isolate ourselves
```

Comment:
Why aren't you catching the cloned value? (and using that to operate the commits further?)